bokeh/models/sources.py

```diff
@@ -64,7 +64,7 @@ class ColumnDataSource(DataSource):
 
     """
 
-    data = Dict(String, Seq(Any), help="""
+    data = ColumnData(String, Seq(Any), help="""
```

Comment:
some sort of `ColumnArray` type instead of `Seq(Any)` might allow us to use a to/from_json override in `ColumnArray` instead of the `isinstance(item_type, Any)` thing. Or `ColumnData` could also have a `from_json` that knows when to bypass the `Seq` from_json, so standalone `Seq(Any)` outside of `ColumnData` would be unaffected.